R-packages/covidcast/R/covidcast.R

```diff
@@ -628,38 +626,49 @@ summary.covidcast_meta = function(object, ...) {
 
 ##########
 
-# Helper function, not user-facing, to fetch a single geo-value.
-# covidcast_signal can then loop over multiple geos to produce its result.
-single_geo <- function(data_source, signal, start_day, end_day, geo_type,
+# Helper function, not user-facing, to loop through a sequence of days, call
+# covidcast for each one and combine the results
+covidcast_days <- function(data_source, signal, start_day, end_day, geo_type,
                        geo_value, as_of, issues, lag) {
   ndays <- as.numeric(end_day - start_day)
   dat <- list()
 
   # The API limits the number of rows that can be returned at once, so we query
   # each day separately.
   for (i in seq(ndays + 1)) {
-    day <- date_to_string(start_day + i - 1)
+    query_day <- start_day + i - 1
+    day_str <- date_to_string(query_day)
     dat[[i]] <- covidcast(data_source = data_source,
                           signal = signal,
                           time_type = "day",
                           geo_type = geo_type,
-                          time_values = day,
+                          time_values = day_str,
                           geo_value = geo_value,
                           as_of = as_of,
                           issues = issues,
                           lag = lag)
     message(sprintf("Fetched day %s: %s, %s, num_entries = %s",
-                    day,
+                    query_day,
                     dat[[i]]$result,
                     dat[[i]]$message,
                     nrow(dat[[i]]$epidata)))
-
-    if (dat[[i]]$message != "success") {
-      warn(paste0("Fetching ", signal, " from ", data_source, " for ", day,
-                  " in geography '", geo_value, "': ", dat[[i]]$message),
+    if (dat[[i]]$message == "success") {
+      returned_geo_values <- dat[[i]]$epidata$geo_value
+      missed_geos <- setdiff(geo_value, returned_geo_values)
+      if (length(missed_geos) > 0) {
+        missed_geos_str <- paste0(missed_geos, collapse = ", ")
+        warn(message =
+               sprintf("Data not fetched for some geographies on %s: %s",
+                       query_day, missed_geos_str)
+             )
```

Comment:
Just like the `warn` call on line 666, could you augment this one with extra parameters stored in the warning object? Plus a distinct class, like `covidcast_missing_geos` or something?

See #63 for motivation. The forecaster code will be able to catch the warning, extract appropriate data from it, and log the problem so the forecasters know what is missing.

Reply:
Done.